code/modules/scripting/Interpreter/Interpreter.dm

```diff
@@ -184,6 +184,8 @@ Runs each statement in a block of code.
 			else
 				RaiseError(new/datum/runtimeError/UnknownInstruction())
 
+			CHECK_TICK

```

Comment:
This should keep scripts from hanging the server while they're actually running on each radio message.